odoo/addons/base/models/qweb.py

```diff
@@ -840,822 +629,545 @@ def _compile_static_node(self, el, options):
             # want changes done in deeper recursion to bevisible in earlier ones, we'll pass
             # a copy before continuing the recursion and restore the original afterwards.
             original_nsmap = dict(options['nsmap'])
+
+        if unqualified_el_tag != 't':
+            attributes = ''.join(f' {self._compile_str_html(name)}=\\"{self._compile_str_html(pycompat.to_text(value))}\\"'
+                                for name, value in attrib.items())
+            self._appendText(f'''<{self._compile_str_html(el_tag)}{attributes}''', options)
+            if unqualified_el_tag in self._void_elements:
+                self._appendText('/>', options)
+            else:
+                self._appendText('>', options)
+
+        if el.nsmap:
             options['nsmap'].update(el.nsmap)
-            content = self._compile_directive_content(el, options)
+            body = self._compile_directive_content(el, options, indent)
             options['nsmap'] = original_nsmap
-
-        if unqualified_el_tag == 't':
-            return content
-        tag = u'<%s%s' % (el_tag, u''.join([u' %s="%s"' % (name, escape(pycompat.to_text(value))) for name, value in attrib.items()]))
-        if unqualified_el_tag in self._void_elements:
-            return [self._append(ast.Str(tag + '/>'))] + content
         else:
-            return [self._append(ast.Str(tag + '>'))] + content + [self._append(ast.Str('</%s>' % el_tag))]
+            body = self._compile_directive_content(el, options, indent)
+
+        if unqualified_el_tag != 't':
+            if unqualified_el_tag not in self._void_elements:
+                self._appendText(f'</{self._compile_str_html(el_tag)}>', options)
 
-    def _compile_static_attributes(self, el, options):
-        """ Compile the static attributes of the given element into a list of
-        pairs (name, expression AST). """
+        return body
+
+    def _compile_attributes(self, options, indent):
+        body = self._flushText(options, indent)
+        body.append(self._indent(dedent("""
+            attrs = self._post_processing_att(tagName, attrs, compile_options)
+            for name, value in attrs.items():
+                if value or isinstance(value, str):
+                    yield u' '
+                    yield name
+                    yield u'="'
+                    yield self._compile_str_html(to_text(value))
+                    yield u'"'
```

Comment:
Why not emit a single yield using an f-string or something? (also unnecessary `u` prefixes)